cli/src/pixl_cli/_docker_commands.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import shutil
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+import click
+from loguru import logger
+
+ALLOWED_PROJECT_NAMES = ["pixl_dev", "pixl_test", "pixl_prod"]
+PIXL_ROOT = Path(__file__).parents[3].resolve()
+
+docker_project_option = click.option(
+    "-p",
+    "--project",
+    type=click.Choice(ALLOWED_PROJECT_NAMES, case_sensitive=False),
+    default="pixl_dev",
+    show_default=True,
+    help="Project to run the service for",
+)
+docker_env_option = click.option(
+    "--env-file",
+    type=click.Path(exists=True),
+    multiple=True,
+    default=[".env"],
+    show_default=True,
+    help="Path to the .env file to use with docker compose",
+)
+docker_extra_args = click.argument("extra_args", nargs=-1, type=click.UNPROCESSED)
+
+
+# Required to allow passing unkown options to docker-compose
+# https://click.palletsprojects.com/en/8.1.x/advanced/#forwarding-unknown-options
+@click.command(context_settings={"ignore_unknown_options": True})
+@docker_project_option
+@docker_env_option
+@docker_extra_args
+def up(project: str, env_file: Path, *, extra_args: list) -> None:
+    """Start all the PIXL services"""
+    # Construct the docker-compose arguments
+    docker_args = ["up", "--wait", "--detach", "--build", "--remove-orphans", *extra_args]
+    run_docker_compose(project, env_file, docker_args, working_dir=PIXL_ROOT)
+
+
+# Required to allow passing unkown options to docker-compose
+# https://click.palletsprojects.com/en/8.1.x/advanced/#forwarding-unknown-options
+@click.command(context_settings={"ignore_unknown_options": True})
+@docker_project_option
+@docker_env_option
+@docker_extra_args
+def down(project: str, env_file: Path, *, extra_args: list) -> None:
+    """Stop all the PIXL services"""
+    # Construct the docker-compose arguments
+    docker_args = ["down", *extra_args]
+    run_docker_compose(project, env_file, docker_args, working_dir=PIXL_ROOT)
+
+
+def run_docker_compose(
+    project: str, env_file: Path, args: list, working_dir: Optional[Path]
+) -> None:
+    """Wrapper to run docker-compose through the CLI."""
+    docker_compose_cmd = shutil.which("docker-compose")
```

Comment:
could be wrong but I think `docker-compose` is v1, and we've got v2 on the GAEs which is `docker compose`. 

Also, why is the command not just `docker compose` and let the system to work out where the executable is? You'd get a similar output if its not found in the path and just adds extra steps

Reply:
Yeah the `shutil.which()` is because originally I used `subprocess.Popen`, [which recommends doing this](https://docs.python.org/3/library/subprocess.html#subprocess.Popen), but for `subprocess.run()` it might be fine not to do it?